mule-extensions-api/src/main/java/org/mule/runtime/extension/api/declaration/type/ExtensionObjectTypeHandler.java

```diff
@@ -111,7 +112,10 @@ public ExtensionObjectTypeHandler(ObjectFieldHandler fieldHandler) {
 
       Stereotype stereotype = currentClass.getAnnotation(Stereotype.class);
       handleStereotype(currentClass, annotatedBuilder, allowTopLevelDefinition, stereotype);
+
+      enrichWithTypeAnnotation(currentClass, annotatedBuilder);
```

Comment:
Are the types we want to enrich always object types?

Reply:
This code is for enriching static pojos. There are other enrichers for enriching other things such as  parameters, connection providers, etc. No support for enriching json types in the MVP if that's what you mean